tests/test_pbs.py

```diff
@@ -145,34 +151,18 @@ def test_pbs_config_parameters(self):
         )
 
     def test_default_pbs_config(self):
-        """Success case: if the pbs_config is empty, use the default values."""
-        assert render_job_script(
-            project="tm70",
-            config_path="/path/to/config.yaml",
-            modules=["foo", "bar", "baz"],
-            skip_bitwise_cmp=True,
-            benchcab_path="/absolute/path/to/benchcab",
-            pbs_config={},
-        ) == (
-            f"""#!/bin/bash
-#PBS -l wd
-#PBS -l ncpus={internal.FLUXSITE_DEFAULT_PBS["ncpus"]}
-#PBS -l mem={internal.FLUXSITE_DEFAULT_PBS["mem"]}
-#PBS -l walltime={internal.FLUXSITE_DEFAULT_PBS["walltime"]}
-#PBS -q normal
-#PBS -P tm70
-#PBS -j oe
-#PBS -m e
-#PBS -l storage=gdata/ks32+gdata/hh5
-
-module purge
-module load foo
-module load bar
-module load baz
-
-set -ev
-
-/absolute/path/to/benchcab fluxsite-run-tasks --config=/path/to/config.yaml 
-
-"""
-        )
+        """Failure case: if any key(s) of pbs_config is/are empty, fail the test."""
+        pbs_missing_keys = ["mem", "ncpus", "storage"]
+        with pytest.raises(
+            ValueError,
+            match="Default pbs parameters missing: "
+            + re.escape(str(pbs_missing_keys)),
+        ):
+            render_job_script(
+                project="tm70",
+                config_path="/path/to/config.yaml",
+                modules=["foo", "bar", "baz"],
+                skip_bitwise_cmp=True,
+                benchcab_path="/absolute/path/to/benchcab",
+                pbs_config={"walltime": "48:00:00"},
+            )
```

Comment:
Not that's not the behaviour we want. The test for partial definition of the PBS entries can be done in the tests for the configuration.

Reply:
So I'll add unit tests in `test_config.py` which checks that the different optional parameters have been added (Also check my comment on `pbs.py` on why I changed that specific test

Reply:
I guess what tripped me is we don't have any checks for the other required entries for `render_job_script` and that's because they are not dictionaries. 

In this case, I think I would prefer to use Python `typing` to achieve the same:

1. Define a TypedDict class in `pbs.py`:
```
from typing import TypedDict

class PBSconfig(TypedDict):
    ncpus: int
    mem: str
    walltime: str
```

2. Use to TypedDict the type of the argument of `render_job_script`:
```
def render_job_script(
    project: str,
    config_path: str,
    modules: list,
    benchcab_path: str,
    pbs_config: PBSconfig,
    verbose=False,
    skip_bitwise_cmp=False,
) -> str:
```

(not tested in anyway, so adapt if not working straight out the box)

This would get rid of the exception in the function and the test.

Reply:
I'm not sure if pbs.py is the best place to put the `PBSconfig` definition... But I'm not sure where to put it really.

Reply:
We have various options here:
1. Create a new file related to just `types` define there
2. Define in `internal.py` - not the best place, but current dependency chain of imports would be a tree
3. If we define it in `pbs.py` there could be a circular dependency because `internal.py` also uses the same type to define `FLUXSITE_DEFAULT_PBS`

Reply:
Yes, option 1 is the best.

Reply:
For now, I can define it in `config.py`